ls1intum/Artemis
# `Development`: Refactor course score CSV export

### Checklist
#### General



#### Client

### Motivation and Context
The course scores component had huge methods and was not very understandable.

### Description
Splits up the large methods into smaller ones where it makes sense to make the code more readable. Also reduces duplication by extracting some common functionality when putting values into the rows into their own class. 

### Steps for Testing
Prerequisites:
- 1 Instructor
- Some course with a few exercises and some student submissions.

1. Export and download the course scores CSV file (reachable via the Course Management → turquoise ‘Scores’ button) **before** deploying this PR to a test server.
2. After deployment, download the CSV file again for the same course. It should be identical (or no longer be broken as for example for Georg Willer’s test course on TS1). 

### Review Progress
#### Code Review
#### Manual Tests

### Test Coverage
| File | Branch | Statement |
| --- | --- | --- |
| `course-scores.component.ts` | 73.6% | 94.2% | 
| `course-scores-csv-row-builder.ts` | 100% | 100% |

### Screenshots
n/a
